lib/astutils.cpp

```diff
@@ -3381,7 +3381,7 @@ bool isGlobalData(const Token *expr, bool cpp)
     return globalData || !var;
 }
 
-bool isSizeOfEtc(const Token *tok)
+bool isUnevaluated(const Token *tok)
 {
-    return Token::Match(tok, "sizeof|typeof|offsetof|decltype|__typeof__ (");
+    return Token::Match(tok, "alignof|_Alignof|_alignof|__alignof|__alignof__|decltype|offsetof|sizeof|typeid|typeof|__typeof__ (");
```

Comment:
This should be `tok->previous()` so we can also skip template parameters as well.

Reply:
How will that work at every call site? Do you mean something like `x<decltype(y)>`?

Reply:
I mean something like `A<x == y>`.

Reply:
Okay, but none of the code was even trying to do that, so I'd rather not add it in this PR.